woocommerce/woocommerce
# Add support for appending a unique string to the filename for the wp/v2/media endpoint

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/41643 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the new product editor at /wp-admin/admin.php?page=wc-settings&tab=advanced&section=features
2. Go to /wp-admin/admin.php?page=wc-settings&tab=products&section=downloadable
3. Check the option "Append a unique string to filename for security" and save.
4. Go to Products > Add 
5. Go to the Downloads section
6. Drop a supported file for upload
7. Check that a unique string was appended to the filename.
8. Click on "Add new" > "Upload"
9. Select a supported file
10. Check that a unique string was appended to the filename.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
